### DISCUSSION TITLE
Backup of Media Files not performed daily

### DISCUSSION THREAD

**Discussion by Fluxus00** [inventree/InvenTree#5078] [Q&A]
Hello guys, 

i activated the automatic backup option in the InvenTree configuration. The database backup is working fine, a new backup is created daily, but the backup of the media files does not seem to work as intended. In the past week only one file, containing the media files, had been created. So my question is now, is there an option/ special condition i am missing, or is it working as intended?

Best regards

**Comment by SchrodingersGat** [comment]
The media files backup runs *after* the database backup, and it might not be succeeding.

If you go to admin page, check out if there is a setting value called `_RUN_BACKUP_SUCCESS` (and `_RUN_BACKUP_ATTEMPT`):

![image](https://github.com/inventree/InvenTree/assets/10080325/56e5e0cb-d390-43f4-92c2-f6260c8aebdb)

If you have `_RUN_BACKUP_ATTEMPT` but *not* `_RUN_BACKUP_SUCCESS` then I'd say the process is taking too long and the worker is timing out?

**Reply by Fluxus00** [reply]
Thank you for the reply, it is exactly what you have suggested. I have  _RUN_BACKUP_ATTEMPT but no _RUN_BACKUP_SUCCESS. Is there anything i could do to fix that?

**Reply by SchrodingersGat** [reply]
> In the past week only one file, containing the media files, had been created.

How big is that file?

**Reply by Fluxus00** [reply]
It is around 2.5 GB big, I was surprised how big it really is. So there is no wonder the worker is not able to keep up.

**Reply by SchrodingersGat** [reply]
Yes, the default worker timeout is 30 seconds, so no wonder...

@matmair any thoughts on this? The inbuild backup solution might not cut it here

**Reply by matmair** [reply]
I have seen larger instances work so maybe it is more the drive speed? The easiest solution would be to increase the worker timeout.

**Reply by Fluxus00** [reply]
I have some worries about increasing the worker timeout, as far as I know the only option which can manage the worker timeout is in the gunicorn.conf.py , this configuration would affect also other tasks and could cause some unpleasant side effects. So is there a possibility to increase the worker timeout only for the backup?

**Reply by SchrodingersGat** [reply]
The background worker timeout is separate to the gunicorn worker timeout:

https://github.com/inventree/InvenTree/blob/7f5e8443672695bd8d58507e3a6907bc4c19d304/InvenTree/InvenTree/settings.py#L668

**Reply by Fluxus00** [reply]
thank you. I implemented the change in a test environment, i will inform you about the result. But correct me if i am wrong, this setting seems to affect all background worker / tasks, are there any negative consequences to be expected if the timeout timer is set to 10 minutes or higher?  For comparison, a manual backup with invoke backup took me about 5 minutes.

**Reply by SchrodingersGat** [reply]
AFAIK the only potential downside is if there's some bad code which gets stuck in a loop, and keeps the workers busy for 10 minutes.

**Reply by Fluxus00** [reply]
So the automatic backup worked and so far i have not encountered  a serious issue with the increased worker timeout. It seems like it solved the issue. 
So thank you very much for your help.

**Comment by Fluxus00** [comment]
Okay so i it seems like, the daily backup was only performed correctly for one day, after that it did not work. I am open for suggestions, are there other possible settings that i could change?

**Reply by SchrodingersGat** [reply]
> Okay so i it seems like, the daily backup was only performed correctly for one day, after that it did not work. 

Can you elaborate on "did not work"? Same symptoms as before, or something new?

**Reply by Fluxus00** [reply]
Yes same symptoms, as far as i can tell, it is the same problem/ situation like before i increased the worker timeout

**Reply by SchrodingersGat** [reply]
@Fluxus00 I have started [a new issue](https://github.com/inventree/InvenTree/issues/5108) for further discussion
